modulus/models/fcn_mip_plugin.py

```diff
@@ -165,8 +167,10 @@ def prepare_input(self, input, time):
         data = self.input_map_wts.S.values
         M = torch.sparse_coo_tensor(np.array((i, j)), data).type(dtype).to(device)
 
-        input = input.reshape(input.shape[0], input.shape[1], input.shape[2], -1) @ M.T
-        input = input.reshape(input.shape[0], input.shape[1], input.shape[2], 6, 64, 64)
+        N, T, C = input.shape[0], input.shape[1], input.shape[2]
+        input = input.reshape(N * T * C, -1) @ M.T
+        S = int((M.shape[0] / 6) ** 0.5)
```

Comment:
So this is a fine choice but maybe why can't we have the size of the grid be defined in the xarray dataset file `self.input_map_wts` and `self.output_map_wts`?

Reply:
Yeah, let's retain the current way for now, but once we have some conventions on the map files, we can revisit this.